core/domain/state_domain.py

```diff
@@ -1925,6 +1928,341 @@ def validate(self) -> None:
+        ):
+            raise utils.ValidationError(
+                'Math tag \'raw_latex-with-value\' attribute should not '
+                'be empty.'
+            )
+
+        if 'svg_filename' not in math_content_list:
+            raise utils.ValidationError(
+                'Math tag does not have \'svg_filename-with-value\' '
+                'attribute.'
+            )
+
+        if math_content_list['svg_filename'] is None:
+            raise utils.ValidationError(
+                'Math tag \'svg_filename-with-value\' attribute should not '
+                'be None.'
+            )
+
+        if (
+            math_content_list['svg_filename'].strip()
+            in ('&quot;&quot;', '', '\'\'', '\"\"')
+        ):
+            raise utils.ValidationError(
+                'Math tag \'svg_filename-with-value\' attribute should not '
+                'be empty.'
+            )
+
+        if math_content_list['svg_filename'].strip()[-4:] != '.svg':
+            raise utils.ValidationError(
+                'Math tag \'svg_filename-with-value\' attribute should '
+                'have svg extension.'
+            )
+
+    if is_tags_nested_inside_tabs_or_collapsible:
+        tabs_tags = soup.find_all('oppia-noninteractive-tabs')
+        if len(tabs_tags) > 0:
+            raise utils.ValidationError(
+                'Tabs tag should not be present inside another '
+                'Tabs or Collapsible tag.'
+            )
+
+        collapsible_tags = soup.find_all('oppia-noninteractive-collapsible')
+        if len(collapsible_tags) > 0:
+            raise utils.ValidationError(
+                'Collapsible tag should not be present inside another '
+                'Tabs or Collapsible tag.'
+            )
+
+
+def validate_tabs_and_collapsible_rte_tags(html: str) -> None:
+    """Validates `Tabs` and `Collapsible` RTE tags
+
+    Args:
+        html: str. The RTE content of the state.
+
+    Raises:
+        utils.ValidationError. Raises the validation error in case the
+            `Tabs` or `Collapsible` tag is not valid.
+    """
+    soup = bs4.BeautifulSoup(html, 'html.parser')
+
+    tabs_tags = soup.find_all('oppia-noninteractive-tabs')
+    for tag in tabs_tags:
+        try:
+            tab_content_json = html_validation_service.unescape_html(
+                tag['tab_contents-with-value'])
+
+            tab_content_list = json.loads(tab_content_json)
+
+            if len(tab_content_list) == 0:
+                raise utils.ValidationError(
+                    'No tabs are present inside the tabs tag.'
+                )
+            for tab_content in tab_content_list:
+                validate_rte_tags(
+                    tab_content['content'],
+                    is_tags_nested_inside_tabs_or_collapsible=True
+                )
+        except Exception:
+            raise utils.ValidationError(
+                'No content attribute is present inside the tabs tag.'
+            )
```

Comment:
This try except block is not correct. If the code reaches line 2210 and raises an error, then it will enter the `except Exception` block at line 2218 and raise that error instead of the one at 2210. Just check if 'tabs_contents-with-value' is in 'tag' first.

Reply:
Right, updated!